cypress/integration/26-register_registration/creator_begins_registering_a_registration/begin_registering_registration.js

```diff
@@ -0,0 +1,148 @@
+import { Given, When, Then, And } from 'cypress-cucumber-preprocessor/steps';
+import { USER_WITH_AUTHOR } from '../../../support/constants';
+
+const fileName = 'example.txt';
+const dlrLink = 'https://dlr.unit.no/resources/66888570-3504-4d12-81a4-c3ffe0605945';
+const dcIdentifierLink = 'https://loar.kb.dk/handle/1902/1674?show=full';
+const dcTermLink = 'https://ntnuopen.ntnu.no/ntnu-xmlui/handle/11250/2638973';
+const openGraphLink = 'https://www.nrk.no/norge/klimakur-2030_-mer-strom-og-mindre-kjott-kan-fa-norge-i-mal-1.14883788';
+const highwireLink = 'https://link.springer.com/article/10.1007/s13201-020-01350-9';
+const schemaLink = 'https://e24.no/boers-og-finans/i/AdyrPM/syv-av-tangens-11-referanser-deltok-paa-luksusseminaret';
+
+// Feature: Creator begins registering a Registration
+
+// Common steps
+Given('Creator begins registering a Registration', () => {
+  cy.login(USER_WITH_AUTHOR);
+  cy.get('[data-testid=new-registration]').click({ force: true });
+});
+// End common steps
+
+//   @443
+//   Scenario Outline: Creator begins registering a Registration in the Wizard
+And('they have selected {string} for starting the Wizard', (method) => {
+  cy.wrap(method).as('registrationMethod');
+  if (method === 'Link to registration') {
+    cy.get('[data-testid=new-registration-link]').click({ force: true });
+    cy.get('[data-testid=new-registration-link-input]').type(dlrLink);
+    cy.get('[data-testid=doi-search-button]').click({ force: true });
+  } else if (method === 'Upload file') {
+    cy.get('[data-testid=new-registration-file]').click({ force: true });
+    cy.get('input[type=file]').attachFile(fileName);
+  }
+});
+When('they click Start', () => {
+  cy.get('@registrationMethod').then((method) => {
+    if (method === 'Link to registration') {
+      cy.get('[data-testid=registration-link-next-button]').should('be.enabled').click({ force: true });
+    } else if (method === 'Upload file') {
+      cy.get('[data-testid=registration-file-start-button]').should('be.enabled').click({ force: true });
+    }
+  });
+});
+Then('they see the Wizard', () => {
+  cy.get('[data-testid=nav-tabpanel-description]').should('be.visible');
+});
+// Examples:
+//   | Method               |
+//   | Link to registration |
+//   | Upload file          |
+
+//   @226
+//   Scenario: Creator begins registering a Registration
+Given('that the user is logged in', () => {});
+And('they have Role Creator', () => {});
+And('they are on the Start page', () => {
+  cy.login(USER_WITH_AUTHOR);

```

Comment:
skulle ikke denne vært i Given?

Reply:
Alle `And` som kommer etter `Given` er en del av `Given` steget. Siden `cy.login(USER_WITH_AUTHOR)` gjør alle tre stegene så la jeg den inn sist.